clang/lib/Serialization/ASTWriterDecl.cpp

```diff
@@ -2214,7 +2214,7 @@ void ASTWriter::WriteDeclAbbrevs() {
   Abv->Add(BitCodeAbbrevOp(0));                       // TSCSpec
   Abv->Add(BitCodeAbbrevOp(0));                       // InitStyle
   Abv->Add(BitCodeAbbrevOp(BitCodeAbbrevOp::Fixed, 1)); // isARCPseudoStrong
-  Abv->Add(BitCodeAbbrevOp(0));                       // Linkage
+  Abv->Add(BitCodeAbbrevOp(1));                         // Linkage
```

Comment:
@ChuanqiXu9 confirmed that touching this is fine (thank you!). If I guess correctly that this value corresponds to `Linkage::None` we have in some default constructors, we should consider changing that default to `Linkage::Invalid`.

`Linkage::None` being default feels like we computed the linkage and arrived at `Linkage::None` even when it's not the case.

CC @AaronBallman

Reply:
Agreed that those should be swapped to `Invalid` instead of `None` (better done as a separate follow-up patch as it isn't an NFC change). However, this is a change to the serialization format so a new PCH consumed by an older Clang will misbehave. Please check whether `VERSION_MAJOR` (in `ASTBitCodes.h`) has been updated for this release and if it hasn't been, increment it by one.

Reply:
> If I guess correctly that this value corresponds to Linkage::None we have in some default constructors, we should consider changing that default to Linkage::Invalid.

No. It is not the case. Sorry for didn't explaining things clearly. The value here means the linkage for ParmVarDecl according to line [2185](https://github.com/llvm/llvm-project/pull/71049/files#diff-125f472e690aa3d973bc42aa3c5d580226c5c47661551aca2889f960681aa64dR2185). So it is natural that ParmVarDecl doesn't have a linkage.

Reply:
Ohhhh! Thanks. None makes far more sense now. Still a change to the serialization values though, so we need to check whether we should do a version bump or not.

Reply:
> Please check whether VERSION_MAJOR (in ASTBitCodes.h) has been updated for this release and if it hasn't been, increment it by one.

This has been bumped after 17 cut-off at least once: https://github.com/llvm/llvm-project/commit/6fb08d8f558a6f28db7835acdb88cab83aea2eb4

Reply:
Excellent, thank you! Hmmm, we should probably put a comment there like "this value was last bumped for the Clang NN release" so we don't have to do so much work in the future... (Not as part of these changes, just thinking out loud.)